core/src/main/scala/org/apache/spark/sql/SnappyDDLParser.scala

```diff
@@ -761,35 +761,56 @@ case class DeployCommand(
     coordinates: String,
     alias: String,
     repos: Option[String],
-    jarCache: Option[String]) extends RunnableCommand {
+    jarCache: Option[String],
+    restart: Boolean) extends RunnableCommand {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    val jarsstr = SparkSubmitUtils.resolveMavenCoordinates(coordinates, repos, jarCache)
-    if (jarsstr.nonEmpty) {
-      val jars = jarsstr.split(",")
-      val sc = sparkSession.sparkContext
-      val uris = jars.map(j => sc.env.rpcEnv.fileServer.addFile(new File(j)))
-      SnappySession.addJarURIs(uris)
-      Utils.mapExecutors[Unit](sparkSession.sparkContext, () => {
-        ToolsCallbackInit.toolsCallback.addURIsToExecutorClassLoader(uris)
-        Iterator.empty
-      })
-      val deployCmd = s"$coordinates|${repos.getOrElse("")}|${jarCache.getOrElse("")}"
-      ToolsCallbackInit.toolsCallback.addURIs(alias, jars, deployCmd)
+    try {
+      val jarsstr = SparkSubmitUtils.resolveMavenCoordinates(coordinates, repos, jarCache)
+      if (jarsstr.nonEmpty) {
+        val jars = jarsstr.split(",")
+        val sc = sparkSession.sparkContext
+        val uris = jars.map(j => sc.env.rpcEnv.fileServer.addFile(new File(j)))
+        SnappySession.addJarURIs(uris)
+        Utils.mapExecutors[Unit](sparkSession.sparkContext, () => {
+          ToolsCallbackInit.toolsCallback.addURIsToExecutorClassLoader(uris)
+          Iterator.empty
+        })
+        val deployCmd = s"$coordinates|${repos.getOrElse("")}|${jarCache.getOrElse("")}"
+        ToolsCallbackInit.toolsCallback.addURIs(alias, jars, deployCmd)
+      }
+      Seq.empty[Row]
+    } catch {
+      case ex: Throwable => {
```

Comment:
A catch of all Throwables should check for SystemFailure and cancellation. Like:

```scala
  case ex: Throwable =>
    if (ex.isInstanceOf[Error] && SystemFailure.isJVMFailureError(e.asInstanceOf[Error])) {
      SystemFailure.initiateFailure(ex)
      // If this ever returns, rethrow the error. We're poisoned
      // now, so don't let this thread continue.
      throw ex
    }
    Misc.checkIfCacheClosing(ex)
    ...    
```

Reply:
Ok

Reply:
Also in the core classes ( parser here ) we should not use gemfire classes. Right? So you mean dhave a method in ToolsCallbackInitImpl which perhaps can be used everywhere.

Reply:
GemFire classes can be used just fine and lots of code in core does. But if code depends on GemFireCache then it cannot be used directly because it will fail in smart connector (if that piece of code is executed in smart connector too). Here there is no such problem.

Reply:
Agree about checkIfCacheClosing. Not needed.

Reply:
Ok. Thanks. Got confused about gemfire classes.